Source/Core/VideoCommon/Present.cpp

```diff
@@ -114,6 +140,7 @@ bool Presenter::FetchXFB(u32 xfb_addr, u32 fb_width, u32 fb_stride, u32 fb_heigh
   {
     // Game is blanking the screen
     m_xfb_entry.reset();
+    m_xfb_rect = MathUtil::Rectangle<int>();
```

Comment:
Should this be set back to `{0, 0, MAX_XFB_WIDTH, MAX_XFB_HEIGHT}` in the else case?

Reply:
No. Now i'm not fresh on this but I remember this was intentional. `m_xfb_rect` needs to stay as it was unless the game was blanking the screen. My change is just a little polish to make sure the `m_xfb_rect` and `m_xfb_entry` match all times, instead of just resetting one of the two.

Reply:
Unless I'm misunderstanding something, the only time `m_xfb_rect` is set is when the width/height of the frame-buffer are 0 or at initialization.  That means that once it is set to 0, it'll never be set again.  That doesn't make sense does it?

Reply:
the else case actually already takes care of setting `m_xfb_rect`
`m_xfb_entry = g_texture_cache->GetXFBTexture(xfb_addr, fb_width, fb_height, fb_stride, &m_xfb_rect);`
which then calls `GetDisplayRectForXFBEntry()` with it and fills it up.
`m_xfb_rect` is always accessed after checking `m_xfb_entry` for validity. All this small change did is to set `m_xfb_rect` to zero in case `m_xfb_entry` was null. When `m_xfb_entry` is re-allocated (or changes size), `m_xfb_entry` will immediately take its size again.

Reply:
For some reason I read too quickly the other day and thought that `m_xfb_rect` was a new variable that you were only setting in the single case.  I agree this seems fine then.  Thanks for clarifying!